cmd/fpga_crihook/main.go

```diff
@@ -15,420 +15,339 @@
-	args, err := getFpgaConfArgs(devNode)
-	if err != nil {
-		return errors.Wrapf(err, "failed get fpgaconf args for %s", devNode)
-	}
-	args = append(args, bitStream.path)
-	output, err := bitStream.execer.Command(fpgaconf, args...).CombinedOutput()
-	if err != nil {
-		return errors.Wrapf(err, "failed to program AFU %s to socket %s, region %s: output: %s", bitStream.params.afu, bitStream.params.devNum, bitStream.params.region, string(output))
-	}
-	return nil
+// Config defines structure of OCI hook configuration
+type Config struct {
+	OciVersion string `json:"ociVersion"`
+	Process    struct {
+		User struct {
+			UID int `json:"uid"`
+			Gid int `json:"gid"`
+		} `json:"user"`
+		Args        []string `json:"args"`
+		Env         []string `json:"env"`
+		OomScoreAdj int      `json:"oomScoreAdj"`
+	} `json:"process"`
+	Hostname string `json:"hostname"`
+	Mounts   []struct {
+		Destination string   `json:"destination"`
+		Type        string   `json:"type"`
+		Source      string   `json:"source"`
+		Options     []string `json:"options,omitempty"`
+	} `json:"mounts"`
+	Hooks struct {
+		Prestart []struct {
+			Path string   `json:"path"`
+			Args []string `json:"args"`
+			Env  []string `json:"env"`
+		} `json:"prestart"`
+	} `json:"hooks"`
+	Annotations struct {
+		IoKubernetesContainerHash                     string    `json:"io.kubernetes.container.hash"`
+		IoKubernetesContainerName                     string    `json:"io.kubernetes.container.name"`
+		IoKubernetesContainerRestartCount             string    `json:"io.kubernetes.container.restartCount"`
+		IoKubernetesContainerTerminationMessagePolicy string    `json:"io.kubernetes.container.terminationMessagePolicy"`
+		IoKubernetesCriOContainerID                   string    `json:"io.kubernetes.cri-o.ContainerID"`
+		IoKubernetesCriOContainerType                 string    `json:"io.kubernetes.cri-o.ContainerType"`
+		IoKubernetesCriOCreated                       time.Time `json:"io.kubernetes.cri-o.Created"`
+		IoKubernetesCriOLogPath                       string    `json:"io.kubernetes.cri-o.LogPath"`
+		IoKubernetesCriOMetadata                      string    `json:"io.kubernetes.cri-o.Metadata"`
+		IoKubernetesCriOSandboxID                     string    `json:"io.kubernetes.cri-o.SandboxID"`
+		IoKubernetesCriOSandboxName                   string    `json:"io.kubernetes.cri-o.SandboxName"`
+		IoKubernetesCriOSeccompProfilePath            string    `json:"io.kubernetes.cri-o.SeccompProfilePath"`
+		IoKubernetesCriOStdin                         string    `json:"io.kubernetes.cri-o.Stdin"`
+		IoKubernetesCriOStdinOnce                     string    `json:"io.kubernetes.cri-o.StdinOnce"`
+		IoKubernetesCriOTTY                           string    `json:"io.kubernetes.cri-o.TTY"`
+		IoKubernetesPodName                           string    `json:"io.kubernetes.pod.name"`
+		IoKubernetesPodNamespace                      string    `json:"io.kubernetes.pod.namespace"`
```

Comment:
should we drop most of those annotations that are not interesting to us?

Reply:
done